smaug/python/graph.py

```diff
@@ -102,18 +102,31 @@ def get_node(self, node_name):
         return self.graph.nodes[i]
     return None
 
-  def _create_unique_name(self, name):
+  def get_nodes(self):
+    """Return nodes in the graph proto."""
+    return self.graph.nodes
+
+  def create_unique_name(self, name, mark_as_used=True):
     """ Create a unique name for the node.
 
     Args:
       name: The node's name.
     """
-    if name not in self.node_names:
-      self.node_names[name] = 0
+    if name not in self._node_names:
+      self._node_names[name] = 0
       return name
     else:
-      self.node_names[name] += 1
-      return name + "_%d" % self.node_names[name]
+      self._node_names[name] += 1
+      new_name = "%s_%d" % (name, self._node_names[name])
+      # Make sure the new name is not already used.
+      while new_name in self._node_names:
```

Comment:
What you're looking for is a do-while loop in Python (to avoid duplicating the loop body outside the loop):

```python
while True:
  self._node_names[name] +1
  new_name = "%s_%d" % (name, self._node_names[name])
  if not new_name in self._node_names:
    break
```

Reply:
Right. This is much better.